llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -1734,22 +1734,31 @@ UnaryFunction for_each(R &&Range, UnaryFunction F) {
 /// Provide wrappers to std::all_of which take ranges instead of having to pass
 /// begin/end explicitly.
 template <typename R, typename UnaryPredicate>
-bool all_of(R &&Range, UnaryPredicate P) {
-  return std::all_of(adl_begin(Range), adl_end(Range), P);
+constexpr bool all_of(R &&Range, UnaryPredicate P) {
+  for (auto &&Element : Range)
+    if (!P(std::forward<decltype(Element)>(Element)))
+      return false;
+  return true;
```

Comment:
Could you move the STExtras changes to a separate PR? This should be easy to review and land. We will also want some unit tests for these.